src/input.js

```diff
@@ -168,6 +168,17 @@ class MutationHandler {
 			return;
 		}
 
+		// Replace &nbsp; inserted by the browser with normal space.
+		// We want only normal spaces in the model and in the view. Renderer and DOM Converter will be then responsible
+		// for rendering consecutive spaces using &nbsp;, but the model and the view has to be clear.
+		// Other feature may introduce inserting non-breakable space on specific key stroke (for example shift + space).
+		// However then it will be handled outside of mutations, like enter key is.
+		// The replacing is here because it has to be done before `diff` and `diffToChanges` functions, as they
+		// take `newText` and compare it to (cleaned up) view.
+		// It could also be done in mutation observer too, however if any outside plugin would like to
+		// introduce additional events for mutations, they would get already cleaned up version (this may be good or not).
+		mutation.newText = mutation.newText.replace( /\u00A0/g, ' ' );
```

Comment:
You shouldn't modify a property of an object because there can be other listeners interested in this value.